### PR TITLE
feat(permissions): [skp-126] Adds Permissions endpoints

### PR DESCRIPTION
**Pull request by Gardemarin**

### CODE REVIEW

File: apps/api/src/app/models/permissions/permissions.interface.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  user: UserRef;
+  role: string;
+}
+
+export interface CreatePermissionPayload {
+  user: UserRef;
+  role: string;
+}
+
+export interface PermissionPayload {
```
Comment (doncevas) [inline apps/api/src/app/models/permissions/permissions.interface.ts:15]: if in interface only one field don't create external type, it is already string

File: apps/api/src/app/models/permissions/permissions.schema.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    user: {
+      _id: { type: String, ref: ViewModels.USERS_VIEW },
+      name: String,
+    },
+    role: {
+      type: String,
+      max: 100,
+      required: true,
+    },
+    categories: [],
```
Comment (doncevas) [inline apps/api/src/app/models/permissions/permissions.schema.ts:15]: inside array should add type, bellow same

File: apps/api/src/app/models/permissions/permissions.service.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { ViewModels } from '../../helpers/constants';
+import {
+  PermissionsViewModel,
+  CreatePermissionPayload,
+  PermissionPayload,
+} from './permissions.interface';
+
+@Injectable()
+export class PermissionsModelService {
+  @InjectModel(ViewModels.PERMISSIONS_VIEW) private _model!: Model<
```
Comment (doncevas) [inline apps/api/src/app/models/permissions/permissions.service.ts:15]: model should be public

File: apps/api/src/app/models/permissions/dtos/update-permission.dto.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { PermissionPayload } from '../permissions.interface';
+
+export class UpdatePermissionPayloadDto implements PermissionPayload {
+  @ApiProperty()
+  public readonly role: string;
```
Comment (doncevas) [inline apps/api/src/app/models/permissions/dtos/update-permission.dto.ts:6]: don't create dto for one field, you can add directly in controller

File: apps/api/src/app/models/permissions/permissions.schema.ts (hunk 2 above)
Comment (Gardemarin) [reply apps/api/src/app/models/permissions/permissions.schema.ts:15]: Categories and Lessons features don't exist.

**Pull request merged**